pkg/router/controller/router_controller.go

```diff
@@ -300,7 +301,12 @@ func convertEndpointSliceToEndpointSubset(items []discoveryv1beta1.EndpointSlice
 				if items[i].Endpoints[j].Hostname != nil {
 					epa.Hostname = *items[i].Endpoints[j].Hostname
 				}
-				addresses = append(addresses, epa)
+				// A nil Ready condition indicates an unknown state and should be interpreted as ready.
```

Comment:
I'm confused how a `nil` Ready condition would mean we should interpret the endpoint as ready. Are there docs, or anything of that sorts, on this?

Reply:
Ah, turns out there is https://github.com/kubernetes/api/blob/master/discovery/v1beta1/types.go#L116-L123

Reply:
```
// EndpointConditions represents the current condition of an endpoint.
type EndpointConditions struct {
	// ready indicates that this endpoint is prepared to receive traffic,
	// according to whatever system is managing the endpoint. A nil value
	// indicates an unknown state. In most cases consumers should interpret this
	// unknown state as ready. For compatibility reasons, ready should never be
	// "true" for terminating endpoints.
	// +optional
	Ready *bool `json:"ready,omitempty" protobuf:"bytes,1,name=ready"`
```

Reply:
> Ah, turns out there is https://github.com/kubernetes/api/blob/master/discovery/v1beta1/types.go#L116-L123

This is why I added the comment - just some additional clarity/context if you don't look that the docs and just read the code as it is now.